### PR TITLE
Support global routing using Valhalla

### PR DESCRIPTION
**Pull request by HarelM**
This is an initial PR to allow route planning outside of Israel.
This is part of #1932

### CODE REVIEW

File: IsraelHiking.Web/src/application/services/routing.provider.ts [hunk 1]
```diff
@@ -52,6 +57,120 @@ export class RouterService {
...
+        } as any;
+        switch (routinType) {
+            case "Hike":
+                json.costing = "pedestrian";
+                json.costing_options = {
+                    pedestrian: {
+                        "use_ferry": 1,
+                        "use_living_streets": 0.5,
+                        "use_tracks": 0,
+                        "service_penalty": 15,
```
Comment (zstadler) [inline IsraelHiking.Web/src/application/services/routing.provider.ts:86]:
```suggestion
                        "service_penalty": 0,
```

File: IsraelHiking.Web/src/application/services/routing.provider.ts [hunk 2]
```diff
@@ -52,6 +57,120 @@ export class RouterService {
...
+                        "service_factor": 1,
+                        "shortest": false,
```
Comment (zstadler) [inline IsraelHiking.Web/src/application/services/routing.provider.ts:88]:
```suggestion
                        "shortest": true,
```

File: IsraelHiking.Web/src/application/services/routing.provider.ts [hunk 3]
```diff
@@ -52,6 +57,120 @@ export class RouterService {
...
+                        "use_hills": 0.5,
+                        "walking_speed": 5.1,
+                        "walkway_factor": 1,
+                        "sidewalk_factor": 1,
+                        "alley_factor": 2,
+                        "driveway_factor": 5,
+                        "step_penalty": 0,
+                        "max_hiking_difficulty": 1,
```
Comment (zstadler) [inline IsraelHiking.Web/src/application/services/routing.provider.ts:96]:
```suggestion
                        "max_hiking_difficulty": 6,
```

File: IsraelHiking.Web/src/application/services/routing.provider.ts [hunk 4]
```diff
@@ -52,6 +57,120 @@ export class RouterService {
...
+            ],
+        } as any;
+        switch (routinType) {
+            case "Hike":
+                json.costing = "pedestrian";
+                json.costing_options = {
+                    pedestrian: {
+                        "use_ferry": 1,
+                        "use_living_streets": 0.5,
+                        "use_tracks": 0,
```
Comment (zstadler) [inline IsraelHiking.Web/src/application/services/routing.provider.ts:85]:
```suggestion
                        "use_tracks": 0.4,
```

Make tracks slightly less favorable that paths

File: IsraelHiking.Web/src/application/services/routing.provider.ts (hunk 1 above)
Comment (zstadler) [reply IsraelHiking.Web/src/application/services/routing.provider.ts:86]: I think this is irrelevant when "shortest" is used

File: IsraelHiking.Web/src/application/services/routing.provider.ts (hunk 4 above)
Comment (zstadler) [reply IsraelHiking.Web/src/application/services/routing.provider.ts:85]: I think this is irrelevant when "shortest" is used

File: IsraelHiking.Web/src/application/services/routing.provider.ts [hunk 5]
```diff
@@ -52,6 +57,120 @@ export class RouterService {
...
+                };
+                break;
+            case "Bike":
+                json.costing = "bicycle";
+                json.costing_options = {
+                    bicycle: {
+                        "maneuver_penalty": 5,
+                        "country_crossing_cost": 600,
+                        "use_ferry": 1,
+                        "shortest": false,
```
Comment (zstadler) [inline IsraelHiking.Web/src/application/services/routing.provider.ts:110]:
```suggestion
                        "shortest": true,
```

File: IsraelHiking.Web/src/application/services/routing.provider.ts [hunk 6]
```diff
@@ -52,6 +57,120 @@ export class RouterService {
...
+                        "use_ferry": 1,
+                        "ferry_cost": 300,
+                        "use_living_streets": 0.5,
+                        "use_tracks": 1,
+                        "private_access_penalty": 450,
+                        "ignore_closures": false,
+                        "closure_factor": 9,
+                        "service_penalty": 15,
+                        "service_factor": 1,
+                        "exclude_unpaved": 1,
```
Comment (zstadler) [inline IsraelHiking.Web/src/application/services/routing.provider.ts:141]:
```suggestion
                        "exclude_unpaved": 0,
```

File: IsraelHiking.Web/src/application/services/routing.provider.ts [hunk 7]
```diff
@@ -52,6 +57,120 @@ export class RouterService {
...
+                        "shortest": false,
```
Comment (zstadler) [inline IsraelHiking.Web/src/application/services/routing.provider.ts:142]:
```suggestion
                        "shortest": true,
```

File: IsraelHiking.Web/src/application/services/spatial.service.ts [hunk 8]
```diff
@@ -363,4 +363,9 @@ export class SpatialService {
             pixelY: Math.floor((tile.y - Math.floor(tile.y)) * tileSize)
         };
     }
+
+    public static isInIsrael(latlng: LatLngAlt): boolean {
+        return latlng.lat > 29 && latlng.lat < 34 &&
+                latlng.lng > 34 && latlng.lng < 36;
```
Comment (zstadler) [inline IsraelHiking.Web/src/application/services/spatial.service.ts:369]:
```suggestion
        return latlng.lat > 29.37711 && latlng.lat < 33.35091 &&
                latlng.lng > 34.07929 && latlng.lng < 29.37711;
```

Use the [Geofabrik polygon](https://download.geofabrik.de/asia/israel-and-palestine.poly)'s bounds

Comment (HarelM) [reply IsraelHiking.Web/src/application/services/spatial.service.ts:369]: I don't think the last number is correct, right? `latlng.lng < 29.37711`

Comment (HarelM) [reply IsraelHiking.Web/src/application/services/spatial.service.ts:369]: I've edited the suggestion, please review it and let me know if it's correct.

Comment (zstadler) [reply IsraelHiking.Web/src/application/services/spatial.service.ts:369]: The raw numbers are:


  | Min | Max
-- | -- | --
Lat | 29.37711 | 33.35091
Lon | 34.07929 | 35.91531

I fixed the code to reflect that

**Pull request merged**